ReactNativeClient/lib/components/note-body-viewer.js

```diff
@@ -39,6 +39,56 @@ class NoteBodyViewer extends Component {
 		this.isMounted_ = false;
 	}
 
+	onLoadEnd() {
+		setTimeout(() => {
+			if (this.props.onLoadEnd) this.props.onLoadEnd();
+		}, 100);
+
+		if (this.state.webViewLoaded) return;
+
+		// Need to display after a delay to avoid a white flash before
+		// the content is displayed.
+		setTimeout(() => {
+			if (!this.isMounted_) return;
+			this.setState({ webViewLoaded: true });
+		}, 100);
+	}
+
+	componentWillReceiveProps() {
+		// force update whenever props received
+		this.forceUpdate();
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
+		const safeGetNoteProp = (props, propName) => {
+			if (!props) return null;
+			if (!props.note) return null;
+			return props.note[propName];
+		};
+
+		// To address https://github.com/laurent22/joplin/issues/433
+		// If a checkbox in a note is ticked, the body changes, which normally would trigger a re-render
+		// of this component, which has the unfortunate side effect of making the view scroll back to the top.
+		// This re-rendering however is uncessary since the component is already visually updated via JS.
+		// So here, if the note has not changed, we prevent the component from updating.
+		// This fixes the above issue. A drawback of this is if the note is updated via sync, this change
+		// will not be displayed immediately.
+		const currentNoteId = safeGetNoteProp(this.props, 'id');
+		const nextNoteId = safeGetNoteProp(nextProps, 'id');
+
+		if (currentNoteId !== nextNoteId || nextState.webViewLoaded !== this.state.webViewLoaded) return true;
+
+		// If the length of the body has changed, then it's something other than a checkbox that has changed,
+		// for example a resource that has been attached to the note while in View mode. In that case, update.
+		return (`${safeGetNoteProp(this.props, 'body')}`).length !== (`${safeGetNoteProp(nextProps, 'body')}`).length;
+	}
+
+	rebuildMd() {
+		// this.mdToHtml_.clearCache();
+		this.forceUpdate();
+	}
```

Comment:
Where did the new code above come from? This is something I wrote so I'm surprised it appears as new here. Or is it something I've deleted and you reimported? If so, why?